nibabel/cmdline/diff.py

```diff
@@ -0,0 +1,207 @@
+
+        # if these values are different, store them in a dictionary
+        if are_values_different(*values):
+            difference[field] = values
+
+    return difference
+
+
+def get_data_diff(files):
+    """Get difference between md5 values
+
+        Parameters
+        ----------
+        files: list of actual files
+
+        Returns
+        -------
+        list
+          np.array: md5 values of respective files
+        """
+
+    md5sums = [
+        hashlib.md5(np.ascontiguousarray(nib.load(f).get_data(), dtype=np.float32)).hexdigest()
+        for f in files
+    ]
+
+    if len(set(md5sums)) == 1:
+        return []
+
+    return md5sums
+
+
+def display_diff(files, diff):
+    """Format header differences into a nice string
+
+        Parameters
+        ----------
+        files: list of files that were compared so we can print their names
+        diff: dict of different valued header fields
+
+        Returns
+        -------
+        str
+          string-formatted table of differences
+    """
+    output = ""
+    field_width = "{:<15}"
+    value_width = "{:<55}"
+
+    output += "These files are different.\n"
+    output += field_width.format('Field')
+
+    for f in files:
+        output += value_width.format(os.path.basename(f))
+
+    output += "\n"
+
+    for key, value in diff.items():
+        output += field_width.format(key)
+
+        for item in value:
+            item_str = str(item)
+            # Value might start/end with some invisible spacing characters so we
+            # would "condition" it on both ends a bit
+            item_str = re.sub('^[ \t]+', '<', item_str)
+            item_str = re.sub('[ \t]+$', '>', item_str)
+            # and also replace some other invisible symbols with a question
+            # mark
+            item_str = re.sub('[\x00]', '?', item_str)
+            output += value_width.format(item_str)
+
+        output += "\n"
+
+    return output
+
+
+def main(args=None, out=None):
+    """Getting the show on the road"""
+    out = out or sys.stdout
+    parser = get_opt_parser()
+    (opts, files) = parser.parse_args(args)
+
+    nibabel.cmdline.utils.verbose_level = opts.verbose
+
+    if nibabel.cmdline.utils.verbose_level < 3:
+        # suppress nibabel format-compliance warnings
+        nib.imageglobals.logger.level = 50
+
+    assert len(files) >= 2, "Please enter at least two files"
+
+    file_headers = [nib.load(f).header for f in files]
+
+    # signals "all fields"
+    if opts.header_fields == 'all':
+        # TODO: header fields might vary across file types, thus prior sensing would be needed
+        header_fields = file_headers[0].keys()
+    else:
+        header_fields = opts.header_fields.split(',')
+
+    diff = get_headers_diff(file_headers, header_fields)
+    data_diff = get_data_diff(files)
+
+    if data_diff:
+        diff['DATA(md5)'] = data_diff
+
+    if diff:
+        out.write(display_diff(files, diff))
+        raise SystemExit(1)
```

Comment:
Is this now preferred to `sys.exit` (or just returning a return code and having the entry point exit with that code)?

Reply:
I guess it is where python violated its own Zen:
```ipython
In [4]: import this
The Zen of Python, by Tim Peters
...
There should be one-- and preferably only one --obvious way to do it.
...

In [2]: sys.exit?
Docstring:
exit([status])

Exit the interpreter by raising SystemExit(status).
```
so seems to be exactly the same thing...  but it is easier to explain that "we will raise exception and then test in the test that it was raised" ;)